### PR TITLE
Add support for uint8 en/decoding

### PR DESCRIPTION
**Pull request by gballet**

### CODE REVIEW

File: codec.go [hunk 1]
```diff
@@ -66,6 +66,19 @@ func DefineBool[T ~bool](c *Codec, v *T) {
 	HashBool(c.has, *v)
 }
 
+// DefineUint8 defines the next field as a uint64.
```
Comment (karalabe) [inline codec.go:69]: uint64 -> uint8 🙈

File: decoder.go [hunk 2]
```diff
@@ -109,6 +109,25 @@ func DecodeBool[T ~bool](dec *Decoder, v *T) {
...
+func DecodeUint8[T ~uint8](dec *Decoder, n *T) {
+	if dec.err != nil {
+		return
+	}
+	if dec.inReader != nil {
+		_, dec.err = io.ReadFull(dec.inReader, dec.buf[:1])
+		*n = T(dec.buf[0])
+		dec.inRead += 1
+	} else {
+		if len(dec.inBuffer) < 8 {
```
Comment (karalabe) [inline decoder.go:122]: 8 -> 1

**Comment by karalabe** [comment]
You're gonna hate me, but pls add the type to the README's API matrix.

File: codec.go (hunk 1 above)
Comment (gballet) [reply codec.go:69]:
```suggestion
// DefineUint8 defines the next field as a uint8.
```

File: decoder.go (hunk 2 above)
Comment (gballet) [reply decoder.go:122]:
```suggestion
		if len(dec.inBuffer) < 1 {
```

**Review by karalabe — APPROVED** [review]
SGTM

**Pull request merged**